### PR TITLE
feat(CES): add ces alarm template resource

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add ces alarm template resource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add ces alarm template resource
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/ces/' TESTARGS='-run TestAccCesAlarmTemplate_basic' 
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/ces/ -v -run TestAccCesAlarmTemplate_basic -timeout 360m -parallel 4 
=== RUN   TestAccCesAlarmTemplate_basic 
=== PAUSE TestAccCesAlarmTemplate_basic
=== CONT  TestAccCesAlarmTemplate_basic
--- PASS: TestAccCesAlarmTemplate_basic (15.09s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ces       15.137s
```

### CODE REVIEW

File: docs/resources/ces_alarm_template.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  Value options: **>**, **<**, **=**, **>=**, **<=**.
+
+* `value` - (Required, Int) Specifies the alarm threshold.
+
+* `unit` - (Optional, String) Specifies the unit string of the alarm threshold.
+The unit can contain a maximum of 32 characters.
+
+* `count` - (Required, Int) Specifies the number of consecutive triggering of alarms.  The value ranges from 1 to 5.
+
+* `level` - (Optional, Int) Specifies the alarm level. Value options:
```
Comment (Jason-Zhang9309) [inline docs/resources/ces_alarm_template.md]: add  a description of the default value

File: huaweicloud/services/acceptance/ces/resource_huaweicloud_ces_alarm_template_test.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+  policies {
+	namespace           = "SYS.APIG"
+	dimension_name      = "api_id"
+	metric_name         = "req_count_2xx"
+	period              = 1
+	filter              = "average"
+	comparison_operator = "="
+	value               = "10"
+	unit                = "次/分钟"
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/ces/resource_huaweicloud_ces_alarm_template_test.go]: I think using English here is better

File: huaweicloud/services/ces/resource_huaweicloud_ces_alarm_template.go [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the type of the CES alarm template.`,
+			},
+			"association_alarm_total": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `Indicates the total num of the alarm that associated with the alarm template.`,
+			},
+			"delete_associate_alarm": {
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/ces/resource_huaweicloud_ces_alarm_template.go]: missing in doc, and it looks this should be a parameter used in delete func

File: huaweicloud/services/ces/resource_huaweicloud_ces_alarm_template.go [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+	return bodyParams
+}
+
+func buildAlarmTemplatePoliciesChildBody(d *schema.ResourceData) []map[string]interface{} {
+	rawParams := d.Get("policies").([]interface{})
+	if len(rawParams) == 0 {
+		return nil
+	}
+
+	params := make([]map[string]interface{}, 0)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/ces/resource_huaweicloud_ces_alarm_template.go]:
```suggestion
	params := make([]map[string]interface{}, 0, len(rawParams))
```

File: huaweicloud/services/acceptance/ces/resource_huaweicloud_ces_alarm_template_test.go [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testCesAlarmTemplate_basic(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_ces_alarm_template" "test" {
+  name = "%s"
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/ces/resource_huaweicloud_ces_alarm_template_test.go]: it better to add `description` in the test

File: huaweicloud/services/acceptance/ces/resource_huaweicloud_ces_alarm_template_test.go [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testCesAlarmTemplate_basic(name),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(rName, "name", name),
+					resource.TestCheckResourceAttrSet(rName, "type"),
+					resource.TestCheckResourceAttrSet(rName, "association_alarm_total"),
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/ces/resource_huaweicloud_ces_alarm_template_test.go:84]: the policies should also be checked

File: docs/resources/ces_alarm_template.md (hunk 1 above)
Comment (houpeng80) [reply docs/resources/ces_alarm_template.md]: there is no default value, but add an explain

File: huaweicloud/services/acceptance/ces/resource_huaweicloud_ces_alarm_template_test.go (hunk 2 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/ces/resource_huaweicloud_ces_alarm_template_test.go]: done

File: huaweicloud/services/ces/resource_huaweicloud_ces_alarm_template.go (hunk 3 above)
Comment (houpeng80) [reply huaweicloud/services/ces/resource_huaweicloud_ces_alarm_template.go]: done

File: huaweicloud/services/ces/resource_huaweicloud_ces_alarm_template.go (hunk 4 above)
Comment (houpeng80) [reply huaweicloud/services/ces/resource_huaweicloud_ces_alarm_template.go]: done

File: huaweicloud/services/acceptance/ces/resource_huaweicloud_ces_alarm_template_test.go (hunk 5 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/ces/resource_huaweicloud_ces_alarm_template_test.go]: done

File: huaweicloud/services/acceptance/ces/resource_huaweicloud_ces_alarm_template_test.go (hunk 6 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/ces/resource_huaweicloud_ces_alarm_template_test.go:84]: done

**Comment by houpeng80** [comment]
make testacc TEST='./huaweicloud/services/acceptance/ces/' TESTARGS='-run TestAccCesAlarmTemplate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ces/ -v -run TestAccCesAlarmTemplate_basic -timeout 360m -parallel 4
=== RUN   TestAccCesAlarmTemplate_basic
=== PAUSE TestAccCesAlarmTemplate_basic
=== CONT  TestAccCesAlarmTemplate_basic
--- PASS: TestAccCesAlarmTemplate_basic (13.82s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ces       13.863s

File: huaweicloud/services/ces/resource_huaweicloud_ces_alarm_template.go [hunk 7]
```diff
@@ -0,0 +1,383 @@
...
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: `Specifies the number of consecutive triggering of alarms.`,
+			},
+			"suppress_duration": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: `Specifies the alarm suppression cycle.`,
+			},
+			"level": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/ces/resource_huaweicloud_ces_alarm_template.go]: this parameter is called `alarm_level` in ces_alarmrule

File: docs/resources/ces_alarm_template.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+    namespace           = "SYS.APIG"
+    dimension_name      = "api_id"
+    metric_name         = "req_count_2xx"
+    period              = 1
+    filter              = "average"
+    comparison_operator = ">="
+    value               = "10"
+    unit                = "times/minute"
+    count               = 3
+    alarm_level         = 2
```
Comment (ShiChangkuo) [inline docs/resources/ces_alarm_template.md:27]: alarm_level or level?

File: huaweicloud/services/ces/resource_huaweicloud_ces_alarm_template.go (hunk 7 above)
Comment (houpeng80) [reply huaweicloud/services/ces/resource_huaweicloud_ces_alarm_template.go]: it has been change to alarm_level

File: docs/resources/ces_alarm_template.md (hunk 8 above)
Comment (houpeng80) [reply docs/resources/ces_alarm_template.md:27]: use alarm_level

**Pull request merged**